File: starter.py
from llama_index import VectorStoreIndex, SimpleDirectoryReader
from dotenv import load_dotenv
load_dotenv()

document = SimpleDirectoryReader("data").load_data()
print()
print(type(document))
print(document)
document = SimpleDirectoryReader(
            input_files=["./omara.txt"]
        ).load_data()
print()
print(type(document))
print(document)
index = VectorStoreIndex.from_documents(document)
'''
index = VectorStoreIndex.from_documents(document)

query_engine = index.as_query_engine(streaming=True)
response = query_engine.query("In one sentence, explain what a JK Flip Flop is")
print(response)
'''

File: starter.js
import fs from "fs/promises"
import { Document, VectorStoreIndex } from "llamaindex"

async function main() {
    const essay = await fs.readFile(
        './paul_graham_essay.txt',
        'utf-8',
    );
    const document = new Document({ text: essay });
    
    const index = await VectorStoreIndex.fromDocuments([document])
    const queryEngine = index.asQueryEngine();

    const response = await queryEngine.query(
        "What is the author's favorite hobby?",
    );
    console.log(response.toString());
}

main();